src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let tempClone = cloneObj(state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        tempClone = Object.assign(tempClone, action.extraData);
```

Comment:
You can use cloneObj function in this case if you add first parametr with default value = {}